wgpu-core/src/instance.rs

```diff
@@ -242,6 +251,48 @@ impl From<HalDeviceType> for DeviceType {
     }
 }
 
+impl PartialEq<HalDeviceType> for DeviceType {
+    fn eq(&self, other: &HalDeviceType) -> bool {
+        match self {
+            Self::Other => {
+                if let HalDeviceType::Other = other {
```

Comment:
This `if`-`else` can be replaced with the upcoming [`matches!`](https://doc.rust-lang.org/std/macro.matches.html), I think it's landing in `1.42`.